### PR TITLE
MET-32: As Julia, I can display all the measurement families

### PR DESCRIPTION
**Pull request by ValentinMumble**
**Description (for Contributor and Core Developer)**

- As Julia, I can display all the measurement families
image

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | Todo
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/less/pages/Default.less [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
   &-horizontalScrollContainer {
     overflow-x: auto;
-    height: ~"calc(100vh - 246px)";
+    height: ~'calc(100vh - 246px)';
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/less/pages/Default.less]: Unstage this file plzzz

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/bridge/dependencies.ts [hunk 2]
```diff
@@ -1,18 +1,24 @@
-// const router = require('pim/router');
-const translate = require('oro/translator');
+import * as trans from 'pimui/lib/translator';
+const router = require('pim/router');
 const viewBuilder = require('pim/form-builder');
 // const messenger = require('oro/messenger');
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/bridge/dependencies.ts]: shall we clean it ?

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/context/router-context.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import {createContext} from 'react';
+
+export interface RouterContextValue {
```
Comment (ValentinMumble) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/context/router-context.ts]: Use type

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/context/router-context.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  generate: (route: string, parameters?: {[param: string]: string}) => string;
+  redirect: (fragment: string, options?: object) => void;
+}
+
+export const RouterContext = createContext<RouterContextValue>({
+  generate: (route, parameters) => route + (parameters ? '?' + new URLSearchParams(parameters).toString() : ''),
+  redirect: () => undefined,
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/context/router-context.ts]: shouldn't we return an undefined here ? I dk, in this case I would throw an exception ? I don't know :/

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/hooks/use-measurement-families.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import {RouterContext} from 'akeneomeasure/context/router-context';
+
+const fetcher = async (route: string) => {
+  const response = await fetch(route);
+
+  return await response.json();
+};
+
+export const useMeasurementFamilies = () => {
+  const [measurementFamilies, setMeasurementFamilies] = useState<MeasurementFamily[] | null>(null);
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/hooks/use-measurement-families.tsx]: Could we set the default value to empty array here ?

Comment (ValentinMumble) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/hooks/use-measurement-families.tsx]: We use `null` to know when the page is not yet initialized and discrimate the cases between not yet loaded and empty list ;)

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/model/measurement-family.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+export type MeasurementFamily = {
+  code: string;
+  labels: {
+    [locale: string]: string;
+  };
+  standard_unit_code: string;
+  units: Unit[];
+};
+
+export const getMeasurementFamilyLabel = (measurementFamily: MeasurementFamily, locale: LocaleCode) =>
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/model/measurement-family.ts]: suggestion: shouldn't we put exported functions before everything else (like we do in the back for public functions ?)

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/pages/Index.tsx [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+  const __ = useContext(TranslateContext);
+  const [searchValue, setSearchValue] = useState('');
+
+  const [sortColumn, getSortDirection, toggleSortDirection] = useSorting('label');
+
+  const measurementFamilies = useMeasurementFamilies();
+  const locale = useContext(UserContext)('uiLocale');
+
+  const filteredMeasurementFamilies =
+    null === measurementFamilies
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/pages/Index.tsx]: measurementFamilies empty array ?

Comment (SamirBoulil) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/pages/Index.tsx]: seen IRL

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/pages/Index.tsx [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+  const measurementFamiliesCount = null === measurementFamilies ? 0 : measurementFamilies.length;
+  const filteredMeasurementFamiliesCount =
+    null === filteredMeasurementFamilies ? 0 : filteredMeasurementFamilies.length;
+
+  return (
+    <>
+      <PageHeader
+        userButtons={
+          <PimView
+            className="AknTitleContainer-userMenuContainer AknTitleContainer-userMenu"
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/pages/Index.tsx]: Bemable => styled component ?

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/bridge/controller/settings.tsx [hunk 9]
```diff
@@ -17,6 +17,7 @@ class SettingsController extends ReactController {
   renderRoute() {
     mediator.trigger('pim_menu:highlight:tab', {extension: 'pim-menu-settings'});
     mediator.trigger('pim_menu:highlight:item', {extension: 'pim-menu-measurements-settings'});
+    this.$el.css({height: '100vh', overflow: 'auto'});
```
Comment (tseho) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/bridge/controller/settings.tsx]: why ?

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/bridge/dependencies.ts [hunk 10]
```diff
@@ -1,18 +1,24 @@
-// const router = require('pim/router');
-const translate = require('oro/translator');
+import * as trans from 'pimui/lib/translator';
```
Comment (tseho) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/bridge/dependencies.ts]: why ?

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/bridge/dependencies.ts [hunk 11]
```diff
@@ -1,18 +1,24 @@
...
 // const messenger = require('oro/messenger');
-// const userContext = require('pim/user-context');
+const userContext = require('pim/user-context');
 // const securityContext = require('pim/security-context');
 
+const __ = (key: string, placeholders: any = {}, number: number = 1) => {
+  const translation = trans.get(key, {...placeholders}, number);
+
+  return undefined === translation ? key : translation;
+};
```
Comment (tseho) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/bridge/dependencies.ts]: IMHO, should be extracted but stay in bridge

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/shared/components/result-count.tsx [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  white-space: nowrap;
+  color: ${props => props.theme.color.purple100};
+  margin-left: 10px;
+  line-height: 16px;
+  text-transform: none;
+`;
+
+type ResultCountProps = {
+  count: number | null;
+  labelKey?: string;
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/shared/components/result-count.tsx]: can't we say: "labelKey: string | 'measurements.list.result_count' here ?

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/shared/components/result-count.tsx [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+};
+
+export const ResultCount = ({count = null, labelKey}: ResultCountProps) =>
+  count === null ? null : (
+    <Container>
+      {useContext(TranslateContext)(
+        labelKey || 'measurements.list.result_count',
```
Comment (SamirBoulil) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/shared/components/result-count.tsx]: then we get rid of this ?

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/context/legacy-context.ts [hunk 14]
```diff
@@ -1,5 +1,5 @@
 import {createContext} from 'react';
-import {ViewBuilder} from 'akeneomeasure/shared/legacy/pim-view/view-builder';
+import {ViewBuilder} from 'akeneomeasure/shared/components/pim-view/view-builder';
```
Comment (tseho) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/context/legacy-context.ts]: why it has been moved from legacy to shared components ?

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/context/user-context.ts [hunk 15]
```diff
@@ -0,0 +1,5 @@
+import {createContext} from 'react';
+
+export type UserContextValue = (key: string) => string;
+
+export const UserContext = createContext<UserContextValue>(() => '');
```
Comment (tseho) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/context/user-context.ts]: Why are the context not in the bridge anymore ?

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/hooks/use-measurement-families.tsx (hunk 5 above)
Comment (tseho) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/hooks/use-measurement-families.tsx]: I do agree with @SamirBoulil, having the hook returning both measurementFamilies and isLoading could be better.

File: src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/pages/Index.tsx (hunk 8 above)
Comment (ValentinMumble) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Resources/public/pages/Index.tsx]: This is a legacy component, we don't want to redo everything right now

**Pull request merged**